drivers/stepper/ti/drv8424.c

```diff
@@ -0,0 +1,775 @@
+	}
+
+	if (!data->enabled) {
+		return -ENODEV;
+	}
+
+	/* Compute target position */
+	data->target_position = data->actual_position + micro_steps;
+
+	ret = drv8424_start_positioning(dev);
+	if (ret != 0) {
+		LOG_ERR("%s: Failed to begin positioning (error %d)", dev->name, ret);
+		return ret;
+	};
+
+	return 0;
+}
+
+static int drv8424_is_moving(const struct device *dev, bool *is_moving)
+{
+	struct drv8424_data *data = dev->data;
+
+	*is_moving = data->is_moving;
+
+	return 0;
+}
+
+static int drv8424_set_reference_position(const struct device *dev, int32_t position)
+{
+	struct drv8424_data *data = dev->data;
+
+	data->actual_position = position;
+
+	return 0;
+}
+
+static int drv8424_get_actual_position(const struct device *dev, int32_t *position)
+{
+	struct drv8424_data *data = dev->data;
+
+	*position = data->actual_position;
+
+	return 0;
+}
+
+static int drv8424_move_to(const struct device *dev, int32_t position)
+{
+	struct drv8424_data *data = dev->data;
+	int ret;
+
+	if (!data->enabled) {
+		return -ENODEV;
+	}
+
+	data->target_position = position;
+
+	ret = drv8424_start_positioning(dev);
+	if (ret != 0) {
+		LOG_ERR("%s: Failed to begin positioning (error %d)", dev->name, ret);
+		return ret;
+	};
+
+	return 0;
+}
+
+static int drv8424_set_max_velocity(const struct device *dev, uint32_t velocity)
+{
+	struct drv8424_data *data = dev->data;
+
+	data->max_velocity = velocity;
+	return 0;
+}
+
+static int drv8424_run(const struct device *dev, const enum stepper_direction direction,
+		       const uint32_t velocity)
+{
+	const struct drv8424_config *config = dev->config;
+	struct drv8424_data *data = dev->data;
+	int ret;
+
+	if (!data->enabled) {
+		return -ENODEV;
+	}
+
+	/* Set direction pin */
+	int dir_value = (direction == STEPPER_DIRECTION_POSITIVE) ? 1 : 0;
+
+	ret = gpio_pin_set_dt(&config->dir_pin, dir_value);
+	if (ret != 0) {
+		LOG_ERR("%s: Failed to set direction pin (error %d)", dev->name, ret);
+		return ret;
+	}
+
+	/* Lock interrupts while modifying settings used by ISR */
+	int key = irq_lock();
+
+	/* Set data used in counter interrupt */
+	data->constant_velocity = true;
+	data->constant_velocity_direction = direction;
+	data->is_moving = true;
+
+	/* Treat velocity 0 by not stepping at all */
+	if (velocity == 0) {
+		ret = counter_stop(config->counter);
+		if (ret != 0) {
+			LOG_ERR("%s: Failed to stop counter (error %d)", dev->name, ret);
+			data->is_moving = false;
+			goto end;
+		}
+		data->is_moving = false;
+		gpio_pin_set_dt(&config->step_pin, 0);
+		data->step_signal_high = false;
+	} else {
+		ret = drv8424_set_counter_frequency(dev, velocity * 2);
+		if (ret != 0) {
+			LOG_ERR("%s: Failed to set counter frequency (error %d)", dev->name, ret);
+			data->is_moving = false;
+			goto end;
+		}
+
+		/* Start counter */
+		ret = counter_start(config->counter);
+		if (ret != 0) {
+			LOG_ERR("%s: Failed to start counter (error %d)", dev->name, ret);
+			data->is_moving = false;
+			goto end;
+		}
```

Comment:
I believe setting `data->is_moving` at other places can be replaced with 
```suggestion
		} else {
		    data->moving = true;
		}    
```